src/main/kotlin/io/zerobase/smarttracing/features/devices/DevicesDao.kt

```diff
@@ -87,4 +89,49 @@ class DevicesDao @Inject constructor(private val graph: GraphTraversalSource) {
             throw EntityCreationException("Error creating scan relationship between devices.", ex)
         }
     }
+
+    fun recordTestResult(testResult: TestResult): ReportId {
+        val reportId = randomUUID().toString()
+        try {
+            val deviceNode = graph.V(testResult.testedParty.value).getIfPresent() ?: throw InvalidIdException(testResult.testedParty)
+            val reporterNode = graph.V(testResult.reportedBy.value).getIfPresent() ?: throw InvalidIdException(testResult.reportedBy)
+            graph.addV("TestResult").
```

Comment:
for personal understanding: test result vertex has 1)  an edge (reported) to the reporter and 2) an edge (report for) to a device. If we have an id to the reporter, why do we also need an edge to the device node since the reporter should already have devices listed under them? Is this for future proofing in case a reporter switches devices etc? 

I also realized that this tested party and device node distinction is probably necessary for facilities that are reporting vs the tested person.

Reply:
read an explanation in the slack: "support the future concept of someone else reporting a test result - like a lab tech or doctor. the test is linked to the device id that the test was for and also to whoever actually sent us the test result."

so resolving convo